src/autopas/containers/directSum/DirectSum.h

```diff
@@ -295,9 +291,111 @@ class DirectSum : public CellBasedParticleContainer<FullParticleCell<Particle>>
 
   } _cellBorderFlagManager;
 
-  ParticleCell &getCell() { return this->_cells.at(0); };
+  /**
+   * Container specific implementation for getParticle. See ParticleContainerInterface::getParticle().
+   *
+   * @tparam regionIter
+   * @param cellIndex
+   * @param particleIndex
+   * @param iteratorBehavior
+   * @param boxMin
+   * @param boxMax
+   * @return tuple<ParticlePointer, CellIndex, ParticleIndex>
+   */
+  template <bool regionIter>
+  std::tuple<const Particle *, size_t, size_t> getParticleImpl(size_t cellIndex, size_t particleIndex,
+                                                               IteratorBehavior iteratorBehavior,
+                                                               const std::array<double, 3> &boxMin,
+                                                               const std::array<double, 3> &boxMax) const {
+    // first and last relevant cell index
+    const auto [startCellIndex, endCellIndex] = [&]() -> std::tuple<size_t, size_t> {
+      // shortcuts to limit the iterator to only part of the domain.
+      if (not(iteratorBehavior & IteratorBehavior::halo)) {
+        // only owned region
+        return {0, 0};
+      }
+      if (not(iteratorBehavior & IteratorBehavior::owned)) {
+        // only halo region
+        return {1, 1};
+      }
+      if constexpr (regionIter) {
+        // if the region lies fully within the container only look at the owned cell
+        if (utils::ArrayMath::less(this->getBoxMin(), boxMin) and utils::ArrayMath::less(boxMax, this->getBoxMax())) {
+          return {0, 0};
+        }
```

Comment:
Yes and no. Yes - there would be the benefit of skipping the owned cell and directly jumping to the halo cell. No - because if you want to iterate only halo particles you tell this to the iterator via the iterator behavior which also restricts which cells are iterated.

Thus I didn't do anything further here.